### PR TITLE
JVM support of BeanIO, OpenSearch and MailMicrosoftOAuth

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/5753
fixes https://github.com/apache/camel-quarkus/issues/5754
fixes https://github.com/apache/camel-quarkus/issues/5755

### CODE REVIEW

**Comment by jamesnetherton** [comment]
You may need to rebase after https://github.com/apache/camel-quarkus/pull/5768 is merged.

File: extensions-jvm/beanio/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+name: "Camel BeanIO"
+description: "Marshal and unmarshal Java beans to and from flat files (such as CSV, delimited, or fixed length formats)"
+metadata:
+  unlisted: true
+  guide: "https://camel.apache.org/camel-quarkus/latest/reference/extensions/beanio.html"
+  categories:
+  - "integration"
+  status:
+  - "preview"
+  - "deprecated"
```
[the remark this answers is not in the document]
Comment (JiriOndrusek) [reply extensions-jvm/beanio/runtime/src/main/resources/META-INF/quarkus-extension.yaml:33]: you are right, 'deprecated' should not be there (I restored it from the past commit)

**Review by aldettinger — APPROVED** [review]

File: extensions-jvm/beanio/runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (JiriOndrusek) [reply extensions-jvm/beanio/runtime/src/main/resources/META-INF/quarkus-extension.yaml]: removed

**Comment by JiriOndrusek** [comment]
> You may need to rebase after #5768 is merged.

PR is rebased

**Pull request merged**